src/gpuarray_buffer_cuda.c

```diff
@@ -1020,41 +1020,115 @@ static void *call_compiler(const char *src, size_t len, const char *arch_arg,
                             2,
 #endif
                             opts);
-  if (log != NULL) {
-    err2 = nvrtcGetProgramLogSize(prog, &buflen);
-    if (err2 != NVRTC_SUCCESS) goto end2;
-    buf = malloc(buflen);
-    if (buf == NULL) goto end2;
-    err2 = nvrtcGetProgramLog(prog, (char *)buf);
-    if (err2 != NVRTC_SUCCESS) goto end2;
-    if (log_len != NULL) *log_len = buflen;
-    *log = (char *)buf;
-    buf = NULL;
+  if (nvrtcGetProgramLogSize(prog, &buflen) == NVRTC_SUCCESS) {
+    strb_appends(log, "NVRTC compile log::\n");
+    if (strb_ensure(log, buflen) == 0)
+      if (nvrtcGetProgramLog(prog, log->s+log->l) == NVRTC_SUCCESS)
+        log->l += buflen - 1;
+    strb_appendc(log, '\n');
   }
-end2:
-  if (err != NVRTC_SUCCESS) goto end;
 
   err = nvrtcGetPTXSize(prog, &buflen);
   if (err != NVRTC_SUCCESS) goto end;
 
-  buf = malloc(buflen);
-  if (buf == NULL) {
-    nvrtcDestroyProgram(&prog);
-    FAIL(NULL, GA_MEMORY_ERROR);
+  if (strb_ensure(ptx, buflen) == 0)
+    err = nvrtcGetPTX(prog, ptx->s+ptx->l);
+
+end:
+  nvrtcDestroyProgram(&prog);
```

Comment:
Now you always destroy the program. Why? It the past it was only in case of error.